### ISSUE TITLE
[求助][评估coco-cn不符合预期]

### ISSUE THREAD

**Issue by randomtutu** [OFA-Sys/Chinese-CLIP#154]
你好

我最近在对比cnclip和一些其他模型的能力，在coco-cn这个数据集上，用vitb16这个ckpt，在text2image的任务上，一直达不到文章里的效果。我简单的说一下我的方法和结果。

文本：coco-cn的human-written-caption的数据，把每个图片对应的5个caption的第1个caption拎出来当做文本(xxx#0)。
图片：coco-cn的testset，一共1k个图片。
评估计算：加载好vitb16的ckpt，model.eval()模式下，infer图片文本pair对emb，然后算1k个文本、图片的相似矩阵，并求r1\r5\r10。我的结果为0.606 0.86 0.936

我把我代码简单的贴在下面，可以参考一下，我觉得可能是我数据也有可能是计算recall的时候有什么diff，可以麻烦给一份你们coco-cn的数据或者看看这个代码的问题么。

谢谢，祝好！
`def evaluate_recall(model, data, args):

    device = torch.device(args.device)
    dataloader = data.dataloader
    all_image_features, all_text_features = [], []
    with torch.no_grad():
        for i, batch in enumerate(dataloader):
            images, texts = batch

            images = images.to(device=device, non_blocking=True)
            texts = texts.to(device=device, non_blocking=True)

            image_features, text_features, logit_scale = model(images, texts)
            all_image_features.append(image_features)
            all_text_features.append(text_features)
            logit_scale = logit_scale.mean()

        text_features = torch.cat(all_text_features)
        image_features = torch.cat(all_image_features)
        logit_scale = logit_scale
        logits_per_text = (logit_scale * text_features @ image_features.t()).detach().cpu()
        ground_truth = torch.arange(len(image_features)).view(-1, 1)

        ranking = torch.argsort(logits_per_text, descending=True)
        preds = torch.where(ranking == ground_truth)[1]
        preds = preds.detach().cpu().numpy()
        metrics = []
        for k in [1, 5, 10]:
            metrics.append(np.mean(preds < k))
        return metrics`

**Comment by yangapku** [comment]
可以的，您可以邮件联系一下我们哈（ya235025@alibaba-inc.com），附上您通过coco-cn的申请截图，我们会提供我们预处理好的coco-cn数据哈。

**Comment by randomtutu** [comment]
> 可以的，您可以邮件联系一下我们哈（[ya235025@alibaba-inc.com](mailto:ya235025@alibaba-inc.com)），附上您通过coco-cn的申请截图，我们会提供我们预处理好的coco-cn数据哈。

你好，已经发邮件申请，请查看一下，谢谢你的帮助！

祝好！
